api/src/core/generation/BoardGenerator.ts

```diff
@@ -83,8 +83,8 @@ export class BoardGenerator {
             }
             goal.categories.forEach((cat) => {
                 const prev =
-                    this.goalsByCategory[this.categoriesByName[cat].id] ?? [];
-                this.goalsByCategory[this.categoriesByName[cat].id] = [
+                    this.goalsByCategoryId[this.categoriesByName[cat.name].id] ?? [];
```

Comment:
If we have the category object and the the id goalsByCatId map, can't we omit the lookup by name here (and on the next line)?